src/statement-list/converters/expr2statement_list.cpp

```diff
@@ -25,3 +65,219 @@ std::string type2stl(const typet &type, const namespacet &ns)
+    convert_and(expr);
+  else if(ID_or == expr.id())
+    convert_or(expr);
+  else if(ID_xor == expr.id() || ID_notequal == expr.id())
+    convert_xor(expr);
+  else if(ID_equal == expr.id())
+    convert_equal(expr);
+  else if(ID_symbol == expr.id())
+    convert_symbol(expr);
+  else if(ID_not == expr.id())
+    convert_not(expr);
+  else // TODO: support more instructions in expr2stl.
+    return expr2c(expr, ns);
+
+  return result.str();
+}
+
+void expr2stlt::convert_and(const exprt &expr)
+{
+  if(inside_bit_string)
+    convert_multiary_bool(expr.operands(), AND);
+  else
+  {
+    std::vector<exprt> operands = expr.operands();
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool(operands, AND);
+  }
+}
+
+void expr2stlt::convert_or(const exprt &expr)
+{
+  if(inside_bit_string)
+    convert_multiary_bool(expr.operands(), OR);
+  else
+  {
+    std::vector<exprt> operands = expr.operands();
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool(operands, OR);
+  }
+}
+
+void expr2stlt::convert_xor(const exprt &expr)
+{
+  if(inside_bit_string)
+    convert_multiary_bool(expr.operands(), XOR);
+  else
+  {
+    std::vector<exprt> operands = expr.operands();
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool(operands, XOR);
+  }
+}
+
+void expr2stlt::convert_equal(const exprt &expr)
+{
+  std::vector<exprt> operands =
+    instrument_equal_operands(expr.op0(), expr.op1());
+  if(inside_bit_string)
+    convert_multiary_bool(operands, XOR);
+  else
+  {
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool(operands, XOR);
+  }
+}
+
+void expr2stlt::convert_not(const exprt &expr)
+{
+  // If a NOT expression is being handled here it must always mark the
+  // beginning of a new bit string.
+  PRECONDITION(!inside_bit_string);
+
+  if(ID_symbol == expr.op0().id())
+  {
+    // Use AN to load the symbol.
+    is_reference = true;
+    result << AND << NOT_POSTFIX << OPERAND_SEPARATOR;
+    convert(expr.op0());
+    result << LINE_SEPARATOR;
+  }
+  else
+  {
+    // Use NOT to negate the RLO after the wrapped expression was loaded.
+    convert(expr.op0());
+    result << NOT LINE_SEPARATOR;
+  }
+}
+
+void expr2stlt::convert_symbol(const exprt &expr)
+{
+  if(is_reference)
+  {
+    result << REFERENCE_FLAG;
+    is_reference = false;
+  }
+  result << id2string(id_shorthand(expr.get(ID_identifier)));
+}
+
+void expr2stlt::convert_multiary_bool(
+  const std::vector<exprt> &operands,
+  const char operation)
+{
+  for(const exprt &op : operands)
+  {
+    if(op.id() == ID_not)
+    {
+      result << operation << NOT_POSTFIX;
+      convert_bool_operand(op.op0());
+    }
+    else
+    {
+      result << operation;
+      convert_bool_operand(op);
+    }
+  }
+}
+
+void expr2stlt::convert_bool_operand(const exprt &op)
+{
+  if(op.id() == ID_symbol)
+  {
+    is_reference = true;
+    result << OPERAND_SEPARATOR;
+    convert(op);
```

Comment:
You already know that `op` is a symbol, so you should invoke `convert_symbol` (or rather, after applying the refactoring recommended below, `convert(to_symbol_expr(op))`